### PR TITLE
Add scheduling op to helm chart

### LINKED ISSUE #696: Ability to add tolerations and nodeSelector on the scheduler-pugins helm chart
### Area

- [X] Scheduler
- [ ] Controller
- [X] Helm Chart
- [ ] Documents

### Other components

-

### What happened?

Unable to deplpy chart on specific nodes in the cluster. 

### What did you expect to happen?

use the chart as is and deploy it with changing values.yaml

### How can we reproduce it (as minimally and precisely as possible)?

The helm chart is missing nodeSelector, affinity and tolerations. Please assist in updating the deployement to allow for it making is usefull to be used off the shelf. 

### Anything else we need to know?

-

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.2", GitCommit:"5835544ca568b757a8ecae5c153f317e5736700e", GitTreeState:"clean", BuildDate:"2022-09-21T14:33:49Z", GoVersion:"go1.19.1", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.0", GitCommit:"a866cbe2e5bbaa01cfd5e969aa3e033f3282a8a2", GitTreeState:"clean", BuildDate:"2022-08-23T17:38:15Z", GoVersion:"go1.19", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Scheduler Plugins version

<details>
release-1.24
</details>

### PR DESCRIPTION
**Pull request by mochizuki875**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Add ability to change `nodeSelector`, `affinity` and `tolerations` of Pods installed via Helm Chart.

#### Which issue(s) this PR fixes:
Fixes #696

#### Does this PR introduce a user-facing change?

```release-note
Adds ability to change NodeSelector, affinity, and tolerance to Helm Chart.
```

### CODE REVIEW

File: manifests/install/charts/as-a-second-scheduler/templates/deployment.yaml [hunk 1]
```diff
@@ -72,3 +82,12 @@ spec:
...
           name: scheduler-config
+      {{- with .Values.scheduler.nodeSelector }}
+      nodeSelector: {{- toYaml . | nindent 8}}
+      {{- end }}
+      {{- with .Values.scheduler.affinity }}
+      affinity: {{- toYaml . | nindent 8}}
+      {{- end }}
+      {{- with .Values.scheduler.tolerations }}
+      tolerations: {{- toYaml . | nindent 8}}
+      {{- end }}
```
Comment (PiotrProkop) [inline manifests/install/charts/as-a-second-scheduler/templates/deployment.yaml:93]: add newline

**Review by PiotrProkop — COMMENTED** [review]
just a small nit, overall looks good to me.

**Comment by Huang-Wei** [comment]
Could you add a release note in the PR's description? (in the `release-notes` section)

/lgtm
/approve

/hold
for release-notes

**Comment by mochizuki875** [comment]
@Huang-Wei
Thanks, and I've done.
Could you check that?

**Comment by mochizuki875** [comment]
PTAL?

**Comment by Huang-Wei** [comment]
@mochizuki875 could you rebase the PR? (although there is no conflict)

The CI [failure](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_scheduler-plugins/697/pull-scheduler-plugins-unit-test-master/1777747395556151296) is related setup-envtest, which should go away after rebasing.

**Comment by mochizuki875** [comment]
@Huang-Wei 

I've attempted to rebase, but CI failure was not solved.
I've fix `setup-envtest` version in `hack/install-envtest.sh` as I think it related to [the issue](https://github.com/kubernetes-sigs/controller-runtime/issues/2720).

FYI: It seems that the same workaround have been taken for #708 that has not been merged yet.

**Comment by Huang-Wei** [comment]
> I've fix setup-envtest version in hack/install-envtest.sh as I think it related to https://github.com/kubernetes-sigs/controller-runtime/issues/2720.

You're right, I had a bad memory... I encountered this issue in my another project and commented [here](https://github.com/kubernetes-sigs/kubebuilder/issues/2480#issuecomment-1771484494). I thought I fixed in this repo but I didn't...

Thanks a lot for the fix!

File: hack/install-envtest.sh [hunk 2]
```diff
@@ -32,6 +32,6 @@ version=$(cat ${SCRIPT_ROOT}/go.mod | grep 'k8s.io/kubernetes' | grep -v '=>' |
 
 GOPATH=$(go env GOPATH)
 TEMP_DIR=${TMPDIR-/tmp}
-go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+go install sigs.k8s.io/controller-runtime/tools/setup-envtest@release-0.17
```
Comment (Huang-Wei) [inline hack/install-envtest.sh:35]: could you change to 0.15 to align with go.mod:

https://github.com/kubernetes-sigs/scheduler-plugins/blob/cd3e4fb738722a7e07b10e4be93eff7285f39f73/go.mod#L31

Comment (mochizuki875) [reply hack/install-envtest.sh]: Ok, I've fixed it.

**Pull request merged**